src/dft_ldos.cpp

```diff
@@ -22,16 +22,31 @@ using namespace std;
 
 namespace meep {
 
-dft_ldos::dft_ldos(double freq_min, double freq_max, int Nfreq) {
-  if (Nfreq <= 1) {
-    omega_min = (freq_min + freq_max) * pi;
-    domega = 0;
-    Nomega = 1;
+// TODO: overload dft_ldos constructor
+dft_ldos::dft_ldos(double *freqs, int Nfreq) {
+  if (Nfreq < 1) { abort("Nfreq must be at least 1"); }
+  else {
+    Nomega = Nfreq;
+    double om[Nomega];
+    for (int i = 0; i < Nfreq; i++) { om[i] = freqs[i] * 2 * pi; }
+    omegas = om;
```

Comment:
This won't work.  `om` will no longer exist once this scope exits.  You need to allocate on the heap with e.g. `new`